container/providers.bzl

```diff
@@ -0,0 +1,45 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Provider definitions"""
+
+# A provider containing information about a container bundle
+BundleInfo = provider(fields = ["container_images", "stamp"])
+
+# A provider identifying container_flatten rules
```

Comment:
Suggestion:
"A provider containing information exposed by container_flatten rules"
or
"A provider containing information about a ..."
Same for other comments - try to pick one of the two options above, or another you think is best, and use it uniformly for all comments.